### PR TITLE
Use harvest calendar for modular and interim payment contracts

### PR DESCRIPTION
**Pull request by dm-ps**
refs: https://github.com/metasfresh/metasfresh/issues/16534

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateTransitionQuery.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+@Builder
+@Value
+public class FlatrateTransitionQuery
+{
+	@NonNull CalendarId calendarId;
+	@NonNull FlatrateTransition templateFlatrateTransition;
```
Comment (adi-stefan) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateTransitionQuery.java]: Please keep this as a POJO, avoid aggregating objects.

**Review by adi-stefan — CHANGES_REQUESTED** [review]
minor thing, please also check failing cucumber

**Review by adi-stefan — CHANGES_REQUESTED** [review]
lgtm

**Review by adi-stefan — APPROVED** [review]

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateTransition.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@Value
+public class FlatrateTransition
+{
+	@NonNull FlatrateTransitionId id;
+
+	@NonNull CalendarId calendarId;
+	
+	@NonNull Boolean endsWithCalendarYear;
+
+	@NonNull Integer termDuration;
```
Comment (teosarca) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateTransition.java:43]: @dm-ps cc @adi-stefan  next time pls consider using Duration instead

**Pull request merged**